### ISSUE TITLE
Great work, but can you please help?

### ISSUE THREAD

**Issue by Velissel** [ghlin/ego#1]
You must be a master of ygopro and yugioh game!!!

I was trying to utilise ygopro engine to create my card game for fun, however I am lack of c++ knowledge, so I had difficulties to start with.

Then I came across your great work, node-ygopro-core wrapper. Although I got card not found error on initialising a duel, I got the feeling that it should be working, just need to supply data source for cards. But that project was last updated at 3 years ago, so I checked your other works which leads me to this place and seems like this repo is still active.

Basically what I am trying to achieve is that, instead of loading static lua scripts that defines effects on cards, players are able to supply cards with effects selected on the fly before a duel is created to start a game, can you please give me some advise where are the key points I need to be aware of to start from?

Cheers

BTW, I noticed that you might be a multi-language speaker, I do speak mandarin too. 大神，求帮助 :)

**Comment by ghlin** [comment]
> BTW, I noticed that you might be a multi-language speaker, I do speak mandarin too.

Thank god！

> Basically what I am trying to achieve is that, instead of loading static lua scripts that defines effects on cards, players are able to supply cards with effects selected on the fly before a duel is created to start a game, can you please give me some advise where are the key points I need to be aware of to start from?

理论上这一点可以通过[ScriptStore](https://github.com/ghlin/ego/blob/51dca6af55cb733774114a7e5a22be40e5906353/packages/engine-interface/src/index.ts#L9)控制，参考[这里](https://github.com/ghlin/ego/blob/51dca6af55cb733774114a7e5a22be40e5906353/packages/replay-laminate/src/main.ts#L110)

 > I was trying to utilise ygopro engine to create my card game for fun, however I am lack of c++ knowledge, so I had difficulties to start with.

不过目前项目的完成度很低，有许多地方仍然需要参考ygopro的代码（大部分逻辑在gui部分，如来自ocgcore的消息各个字段的含义等等）

> But that project was last updated at 3 years ago, so I checked your other works which leads me to this place and seems like this repo is still active.

虽然还是希望尽量更新，但是由于工作，速度应该会很慢很慢。。。

**Comment by Velissel** [comment]
十分感谢您的指导，我先看看您给的资料

**Comment by Velissel** [comment]
@ghlin 大神，我感觉有些理解了，但是跑不起来，能不能教我一下你这个项目怎么才能跑起来，比如

engine的代码貌似只是ygopro的一部分？
replay-laminate要求的libocgcore.so是怎么编译出来的？
如果用replay-laminate回放的话，能用ygopro的回放文件吗？scripts和card数据库应该放在哪里呢？
duel-client貌似没看到和duel-host交互的代码，是怎么和host交互的？

感谢
